File: 232-implement-queue-using-stacks.cpp
class MyQueue {
    stack<int> input, output;
public:
    MyQueue() {
        
    }

    void connect() {
        while(input.size()) {
            output.push(input.top());
            input.pop();
        }
    }
    
    void push(int x) {
        input.push(x);
    }
    
    int pop() { 
        if (output.empty()) {
            connect();
        }
        int x = output.top();
        output.pop();
        return x;
    }
    
    int peek() {
        if (output.empty()) {
            connect();
        }
        return output.top();
    }
    
    bool empty() {
        return output.empty() && input.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 232-implement-queue-using-stacks.py
class MyQueue(object):
    

    def __init__(self):
        self.input = []
        self.output = []
        
    def push(self, x):
        self.input.append(x)

    def pop(self):
        if(len(self.output)==0): 
            while len(self.input) != 0:
                self.output.append(self.input[-1])
                self.input.pop()

        x = self.output[-1]
        self.output.pop()
        return x

    def peek(self):
        if(len(self.output)==0): 
            while len(self.input) != 0:
                self.output.append(self.input[-1])
                self.input.pop()

        return self.output[-1]

    def empty(self):
        return len(self.output)==0 and len(self.input)==0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.append(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
